core/src/main/java/org/infinispan/manager/DefaultCacheManager.java

```diff
@@ -470,7 +471,7 @@ public EmbeddedCacheManager startCaches(final String... cacheNames) {
       authzHelper.checkPermission(AuthorizationPermission.LIFECYCLE);
       List<Thread> threads = new ArrayList<Thread>(cacheNames.length);
       final AtomicReference<RuntimeException> exception = new AtomicReference<RuntimeException>(null);
-      for (final String cacheName : cacheNames) {
+      for (final String cacheName : new HashSet<String>(Arrays.asList(cacheNames))) {
```

Comment:
The creation of a collection just to initialize a set is a bit pointless. Maybe initialize an empty set, and do `if (set.add(cacheName)) { /*startCache*/ }`